context/app/static/js/components/workspaces/TemplateTagsAutocomplete/TemplateTagsAutocomplete.tsx

```diff
@@ -46,6 +50,12 @@ function TemplateTagsAutocomplete({
         isOptionEqualToValue={(option, value) => option === value}
         tagComponent={TagComponent}
         onChange={(_, value: string[]) => {
+          const addedValue = value.find((tag) => !selectedTags.includes(tag));
+
+          if (addedValue) {
+            onSelectTag?.(addedValue);
```

Comment:
The `onSelectTag`/`onSelectRecommendedTag` props are useful for reusable components; however, since this is a `Template Tags`-specific component, it might be easier to follow the logic if the event tracking was built into the component itself (unless there's different tracking behaviors for `workspaceMessaging.tsx` vs. `TemplateSelectStep`?)